rest-server/engines/src/main/scala/com/actionml/engines/ur/URAlgorithm.scala

```diff
@@ -462,11 +462,11 @@ class URAlgorithm private (
     val numResults = query.num.getOrElse(limit)
 
     // create a list of all query correlators that can have a bias (boost or filter) attached
-    getUserHistMatcher(query).zip(getSimilarItemsMatchers(query)).map { case ((userHistoryMatchers, userEvents), similarItemsMatchers) =>
+    getUserHistMatcher(query).zip(getSimilarItemsMatchers(query)).zip(getItemSetMatchers(query)).map { case (((userHistoryMatchers, userEvents), similarItemsMatchers), itemSetMatchers ) =>
```

Comment:
@andreyshikov Is this the correct way to include getItemSetMatchers now that the function returns a Future[Seq[Matchers]]? I had to change the return value of this function so that I could use getSimilarItemsMatchers for the case when the query has only one item in the itemSet.

Reply:
Concurrently is exactly what we want, item and itemSet queries are possible but are labeled "experimental" and so not really supported. Therefore we expect only one of user, item, itemSet but it should work as expected if all or some are in the query.